### ISSUE TITLE
Add a tool for generating useful bug reports (a step or two past 'ansible --version')

### ISSUE THREAD

**Issue by alikins** [ansible/community#106]
The blurb in the issue boilerplate for request 'ansible --version' output is super useful.

Adding some support for the next step could improve bug reports even more.

for ex `ansible-playbook --issue-info site.yml`

'--issue-info' could include:
- all the 'ansible --version' info
- any non-default config
- controller env info (os, arch, python version, etc)
- what inventory mechanism was used
- what roles were referenced
- module/library paths and versions
- what modules/tasks were referenced
  - what module/tasks were actually invoked

Basically, enough info to figure out the specific folks to ping about the issue (and/or assign, etc).

Stretch goal:
Implement so it dumps the info a format that could be 'replayed' to some degree.
(for ex, checkout the matching version, setup matching config, matching python version, roles versions, etc).

**Comment by bcoca** [comment]
i was planning on `ansible-config dump` to get all the settings for the 'current' config and the 'origin' of where they were set.

**Comment by bcoca** [comment]
Also this might belong in proposals.

**Comment by bcoca** [comment]
https://github.com/ansible/proposals/issues/35
